src/main/kotlin/com/rustyrazorblade/easycassstress/commands/Run.kt

```diff
@@ -411,13 +413,19 @@ class Run(
         val metrics = createMetrics()
 
         // set up the rate limiter optimizer and put it on a schedule
-        var optimizer = RateLimiterOptimizer(rateLimiter, metrics, maxReadLatency, maxWriteLatency)
-        optimizer.reset()
-
-        // Schedule the optimizer to run periodically
-        Timer().schedule(10000, 5000) {
-            optimizer.execute()
-        }
+        val optimizer =
+            if (useOptimizer) {
+                val opt = RateLimiterOptimizer(rateLimiter, metrics, maxReadLatency, maxWriteLatency)
+                opt.reset()
+
+                // Schedule the optimizer to run periodically
+                Timer().schedule(10000, 5000) {
+                    opt.execute()
+                }
```

Comment:
feels dirty, shouldn't it be something like

```
optimizer.start()
```

does it make sense to make the 10k and 5k a config?

Reply:
Yeah, it does.  I was actually working on a different patch to refactor out everything that's hard coded, to address the detekt warnings, when i found the issues fixed in this patch.  Expect a follow up to address this.